Src/PerformanceCollector/Perf.Shared/Implementation/QuickPulse/IQuickPulseServiceClient.cs

```diff
@@ -5,7 +5,7 @@
 
     using Microsoft.ApplicationInsights.Extensibility.Filtering;
 
-    internal interface IQuickPulseServiceClient
+    internal interface IQuickPulseServiceClient : IDisposable
```

Comment:
If Dispose() is .NET version specific (e.g. we need it only for .NET core) - then you can make this inheritance conditional on #ifdef NETSTANDARD13 or similar. This will remove two empty implementations from the clients on another .Net versions.

Reply:
conditional inheritance sounds like a slippery slope. I would like to avoid this